flake8_trio/visitors/helpers.py

```diff
@@ -28,7 +28,7 @@
 
     T = TypeVar("T", bound=Flake8TrioVisitor)
     T_CST = TypeVar("T_CST", bound=Flake8TrioVisitor_cst)
-    T_EITHER = TypeVar("T_EITHER", Flake8TrioVisitor, Flake8TrioVisitor_cst)
+    T_EITHER = TypeVar("T_EITHER", bound=Flake8TrioVisitor | Flake8TrioVisitor_cst)
```

Comment:
I think this `|` syntax requires Python 3.10?

Reply:
I think it's fine due to `from __future__ import annotations`?
But regardless, tests run with <3.10, and I don't think we care about whether `pyright` running on 3.9 passes.

Reply:
🤦 it's fine at runtime because we're in an `if TYPE_CHECKING:` block and so it never actually runs, nevermind.

Reply:
haha I didn't realize that either, https://github.com/snok/flake8-type-checking doing it's job!